File: src/main/kotlin/com/fracturedskies/engine/jeact/scope/ComponentScope.java
package com.fracturedskies.engine.jeact.scope;

import javax.enterprise.util.AnnotationLiteral;
import javax.inject.Scope;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Scope
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface ComponentScope {
  final class Literal extends AnnotationLiteral<ComponentScope> implements ComponentScope {
    public static final Literal INSTANCE = new Literal();
  }
}

File: src/main/kotlin/com/fracturedskies/engine/jeact/scope/ComponentScope.kt
package com.fracturedskies.engine.jeact.scope

import javax.enterprise.util.AnnotationLiteral
import javax.inject.Scope

@Scope
@Retention(value = AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.TYPE)
annotation class ComponentScope {
    companion object {
        class Literal : AnnotationLiteral<ComponentScope>() {
            companion object {
                val INSTANCE = Literal()
            }
        }
    }
}
